include/autocheck/generator_combinators.hpp

```diff
@@ -45,10 +46,11 @@ namespace autocheck {
           pred(pred), gen(gen) {}
 
         result_type operator() (size_t size = 0) {
-          while (true) {
+          for(size_t i{0}; i < std::numeric_limits<size_t>::max(); ++i){
```

Comment:
What is the logic here exactly? I don't think the filter promises to halt if the predicate accepts nothing, and in general, no generator should call a sub-generator (including itself) with a larger size than it was given.